1_fetch/src/my_happy_script_ms_11-12-21.R

```diff
@@ -0,0 +1,196 @@
+    text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
+    
+    dev.off()
+}
+
+### Function 4 - Render_data_diag takes the cleaned/processed data used for the plot and prints the model diagnostics 
+### This function no return
+
+render_data_diag <- function(data, output_path = out_path){
+  
+# Save the model diagnostics
+render_data <- list(pgdl_980mean = filter(data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                    dl_980mean = filter(data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                    pb_980mean = filter(data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                    dl_500mean = filter(data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                    pb_500mean = filter(data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                    dl_100mean = filter(data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                    pb_100mean = filter(data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                    pgdl_2mean = filter(data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                    pb_2mean = filter(data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
+
+template_1 <- 'resulted in mean RMSEs (means calculated as average of RMSEs from the five dataset iterations) of {{pgdl_980mean}}, {{dl_980mean}}, and {{pb_980mean}}°C for the PGDL, DL, and PB models, respectively.
+  The relative performance of DL vs PB depended on the amount of training data. The accuracy of Lake Mendota temperature predictions from the DL was better than PB when trained on 500 profiles 
+  ({{dl_500mean}} and {{pb_500mean}}°C, respectively) or more, but worse than PB when training was reduced to 100 profiles ({{dl_100mean}} and {{pb_100mean}}°C respectively) or fewer.
+  The PGDL prediction accuracy was more robust compared to PB when only two profiles were provided for training ({{pgdl_2mean}} and {{pb_2mean}}°C, respectively). '
+
+whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), render_data ) %>% cat(file = file.path(output_path, 'model_diagnostic_text.txt'))
+
+
+}
+
+
+### RUN ###
+
+downloaded_data_path <- data_extraction(filename_output_csv = csv_filename_output,
+                                        download_data_location = in_path,
+                                        selected_id_sciencebase = sb_id_sciencebase)
+
+eval_data = prep_model_data(downloaded_data_file_path = downloaded_data_path,
+                            cleaned_data_location = out_path)
+
+Plot_model_data(data = eval_data, output_plot_location = out_path)
```

Comment:
The function is defined with a lower-case p as `plot_model_data` so this will not run!